verification/spec/connection-handshake/ICS3Types.tla

```diff
@@ -0,0 +1,296 @@
+----------------------------- MODULE ICS3Types -----------------------------
+
+(***************************************************************************
+
+    This module is part of the TLA+ high-level specification for the
+    IBC Connection Handshake protocol (ICS3).
+    
+    This module includes common domain definitions that other modules will
+    extend.
+
+
+ ***************************************************************************)
+
+EXTENDS Naturals
+
+CONSTANTS MaxHeight,
+          AllConnectionIDs,
+          AllClientIDs
+
+
+(******************************* InitClients *****************************
+     A set of records describing the possible initial values for the
+     clients on a chain.
+     
+     A client record contains the following fields:
+
+     - consensusStates -- a set of heights, each height being a Nat 
+       Stores the set of all heights (i.e., consensus states) that this
+       client observed. At initialization time, the client only observes
+       the first height, so the only possible value for this record is
+       {1}.
+
+     - clientID -- a string
+       The identifier of the client. This is expected as a parameter, since
+       it is a chain-specific field at initialization time.
+       
+     - latestHeight -- a natural number
```

Comment:
As we can compute max out of consensusStates maybe it's better idea having just an operator that would compute it. If I understood well @istoilkovska, it might be more model checker friendly if we have an operator instead of additional field.